demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/ui/screen/seed/view/ConfigureSeedView.kt

```diff
@@ -113,5 +130,88 @@ private fun ConfigureSeedMainContent(
         ) {
             Text(text = stringResource(id = R.string.seed_random))
         }
+
+        Spacer(modifier = Modifier.height(16.dp))
+
+        RestoreWalletSection(zcashNetwork, onExistingWallet)
+    }
+}
+
+@Composable
+@Suppress("ktlint:standard:function-naming")
+private fun RestoreWalletSection(
+    zcashNetwork: ZcashNetwork,
+    onExistingWallet: (PersistableWallet) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier = modifier.fillMaxWidth(), verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        val context = LocalContext.current
+
+        var seedPhraseString by rememberSaveable {
+            mutableStateOf("")
+        }
+        var walletBirthdayString by rememberSaveable {
+            mutableStateOf("")
+        }
+        var requiredDataIsEmpty by rememberSaveable {
+            mutableStateOf(true)
+        }
+
+        fun preValidateInput() {
+            requiredDataIsEmpty = seedPhraseString.isEmpty()
+        }
+
+        fun validateAndRestore() {
+            try {
+                val seedPhrase = SeedPhrase.new(seedPhraseString)
+
+                val blockHeight =
+                    walletBirthdayString.toLongOrNull()?.let { blockHeight ->
+                        BlockHeight.new(zcashNetwork, blockHeight)
+                    } ?: WalletFixture.lastKnownBirthday(zcashNetwork) // fallback to last known valid value
+
+                val wallet =
+                    PersistableWallet(
+                        network = zcashNetwork,
+                        endpoint = LightWalletEndpoint.defaultForNetwork(zcashNetwork),
+                        birthday = blockHeight,
+                        seedPhrase = seedPhrase,
+                        walletInitMode = WalletInitMode.RestoreWallet
+                    )
+                onExistingWallet(wallet)
+            } catch (e: IllegalArgumentException) {
+                Toast.makeText(context, e.message, Toast.LENGTH_LONG).show()
```

Comment:
It's great we provide such feedback to the UI. What about logging the exception with `Twig.error`? Could that also be useful for developers?